### PR TITLE
Fixed exception in Invoke method

### LINKED ISSUE #124: If assemblyName is less than 4 characters, an exception occurs in Latebind.InvokeMethod method.
#### Problem
If assemblyName is less than 4 characters, an exception occurs in Latebind.InvokeMethod method.

- Latebind.InvokeMethod method. 
https://github.com/OpenTouryoProject/OpenTouryo/blob/develop/root/programs/C%23/Frameworks/Infrastructure/Public/Util/Latebind.cs#L85

```C#
85  // アセンブリ ロード 
86  string ext = ".dll"; 
87  int extLen = ext.Length; 
88  if (assemblyName.Substring(assemblyName.Length - extLen, extLen) == ext) 
89  { 
90    //  アセンブリ・ファイルのパス指定の場合、Assembly.LoadFile 
91    mod = Assembly.LoadFile(assemblyName); 
92  } 
```

This exception occurs when you set a value less than 0 to the first argument of String.Substring.

#### Countermeasures
Please to prevent the exception by adding the check processing of assemblyName method.

You can easily fix it by using the EndsWith.

- String.EndsWith(String, StringComparison)  
https://msdn.microsoft.com/ja-jp/library/ms131430.aspx

### PR DESCRIPTION
**Pull request by daisukenishino**
Fixed exception.

fixed #124
If assemblyName is less than 4 characters, an exception occurs in　Latebind.InvokeMethod method.

### CODE REVIEW

File: root/programs/C#/Frameworks/Infrastructure/Public/Util/Latebind.cs [hunk 1]
```diff
@@ -84,8 +85,7 @@ public static object InvokeMethod(string assemblyName, string className, string
             {
                 // アセンブリ ロード
                 string ext = ".dll";
-                int extLen = ext.Length;
-                if (assemblyName.Substring(assemblyName.Length - extLen, extLen) == ext)
+                if (assemblyName.EndsWith(ext))
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Util/Latebind.cs]: I think that you should call ToLowerCase method.

```c#
if (assemblyName.ToLowerCase().EndsWith(ext))
```

Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/Public/Util/Latebind.cs]: @yusukemaegawa 

I see.

@kishoreandukuri 

Please modify the nishino branch directly.
You do not need to use pull request for fix in this time.

Comment (kishoreandukuri) [inline root/programs/C#/Frameworks/Infrastructure/Public/Util/Latebind.cs]: @daisukenishino 
Sure, we will modify he nishino branch directly.

Comment (kishoreandukuri) [inline root/programs/C#/Frameworks/Infrastructure/Public/Util/Latebind.cs]: @daisukenishino 

We have modified the nishino branch directly with [this](https://github.com/OpenTouryoProject/OpenTouryo/pull/140/files#r49158443) chanages.

**Comment by daisukenishino** [comment]
>#### Checkout via command line
If you cannot merge a pull request automatically here, you have the option of checking it out via command line to resolve conflicts and perform a manual merge.
https://github.com/SymphonyTeleca/OpenTouryo.git
##### Step 1: From your project repository, check out a new branch and test the changes.
git checkout -b SymphonyTeleca-nishino develop
git pull https://github.com/SymphonyTeleca/OpenTouryo.git nishino
##### Step 2: Merge the changes and update on GitHub.
git checkout develop
git merge --no-ff SymphonyTeleca-nishino
git push origin develop

Since the above was displayed, I performed same operation as above by TortoiseGit.
And then, this pull request state was change to the "merged commit".

Wonder !!

**Pull request merged**